crates/oxc_linter/src/rules/eslint/id_length.rs

```diff
@@ -0,0 +1,573 @@
+use oxc_ast::AstKind;
+use oxc_ast::ast::BindingPatternKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{ContentEq, GetSpan, Span};
+
+use lazy_regex::Regex;
+use serde_json::Value;
+use unicode_segmentation::UnicodeSegmentation;
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn id_length_is_too_short_diagnostic(span: Span, config_min: u64) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("Identifier name is too short (< {config_min}).")).with_label(span)
+}
+
+fn id_length_is_too_long_diagnostic(span: Span, config_max: u64) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("Identifier name is too long (> {config_max}).")).with_label(span)
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+enum PropertyKind {
+    #[default]
+    Always,
+    Never,
+}
+
+impl PropertyKind {
+    pub fn from(raw: &str) -> Self {
+        if raw == "never" { PropertyKind::Never } else { PropertyKind::default() }
+    }
+}
+
+#[derive(Debug, Default, Clone)]
```

Comment:
I tried to add `JsonSchema` to this, so that the config could be added to declare_oxc_lint!, but `Regex` doesn't have an implement `JsonSchema` and I wasn't sure the best way to do that. Didn't feel appropriate to do that here.